app/src/main/java/fr/free/nrw/commons/depictions/SubClass/SubDepictionListPresenter.java

```diff
@@ -0,0 +1,167 @@
+    }
+    @Override
+    public void onAttachView(SubDepictionListContract.View view) {
+        this.view = view;
+    }
+
+    @Override
+    public void onDetachView() {
+        this.view = DUMMY;
+    }
+
+    /**
+     * Store the current query in Recent searches
+     */
+    @Override
+    public void saveQuery() {
+        RecentSearch recentSearch = recentSearchesDao.find(query);
+
+        // Newly searched query...
+        if (recentSearch == null) {
+            recentSearch = new RecentSearch(null, query, new Date());
+        } else {
+            recentSearch.setLastSearched(new Date());
+        }
+        recentSearchesDao.save(recentSearch);
+    }
+
+    /**
+     * Calls Wikibase APIs to fetch Thumbnail image for a given wikidata item
+     */
+    @Override
+    public void fetchThumbnailForEntityId(String entityId, int position) {
+        compositeDisposable.add(depictsClient.getP18ForItem(entityId)
+                .subscribeOn(ioScheduler)
+                .observeOn(mainThreadScheduler)
+                .timeout(TIMEOUT_SECONDS, TimeUnit.SECONDS)
+                .subscribe(response -> {
+                    view.onImageUrlFetched(response,position);
+                }));
+    }
+
+    @Override
+    public void initSubDepictionList(String qid, Boolean isParentClass) throws IOException {
+        size = 0;
+        if (isParentClass) {
+            compositeDisposable.add(okHttpJsonApiClient.getParentQIDs(qid)
+                    .subscribeOn(ioScheduler)
+                    .observeOn(mainThreadScheduler)
+                    .subscribe(this::handleSuccess, this::handleError));
+        } else {
+            compositeDisposable.add(okHttpJsonApiClient.getChildQIDs(qid)
+                    .subscribeOn(ioScheduler)
+                    .observeOn(mainThreadScheduler)
+                    .subscribe(this::handleSuccess, this::handleError));
+        }
+
+    }
+
+    @Override
+    public String getQuery() {
+        return query;
+    }
+
+    /**
+     * Handles the success scenario
+     * it initializes the recycler view by adding items to the adapter
+     */
+    public void handleSuccess(List<DepictedItem> mediaList) {
+        if (mediaList == null || mediaList.isEmpty()) {
+            if(queryList.isEmpty()){
+                view.initErrorView();
+            }else{
+                view.setIsLastPage(true);
+            }
+        } else {
+            this.queryList.addAll(mediaList);
+            view.onSuccess(mediaList);
+            for (DepictedItem m : mediaList) {
+                fetchThumbnailForEntityId(m.getEntityId(), size++);
+            }
+        }
+    }
+
+    /**
+     * Logs and handles API error scenario
+     */
+    private void handleError(Throwable throwable) {
+        Timber.e(throwable, "Error occurred while loading queried depictions");
+        try {
+            view.initErrorView();
+            view.showSnackbar();
+        } catch (Exception e) {
```

Comment:
IMO we should handle just specific exceptions. Also, use timber logs.

Reply:
We should heavily consider sharing our inspections so eople don't repeat these mistakes and get quick fixes
![image](https://user-images.githubusercontent.com/3358282/77308668-3f1ded00-6cf3-11ea-8be5-d80b103a35af.png)
![image](https://user-images.githubusercontent.com/3358282/77308700-49d88200-6cf3-11ea-8280-bf9c2dcaa641.png)

I believe timber also has a lint rule